src/5.0_to_5.1/migrateSettings.ts

```diff
@@ -0,0 +1,73 @@
+import { ContentRecord } from "@activeviam/activeui-sdk-5.0";
+
+const settingsToPermissionsMapping = {
+  "userFilters.areEnabled": "canUseUserFilters",
+  "userQueryContext.isEnabled": "canUseUserQueryContext",
+  "calculatedMeasures.areEnabled": "canUseCalculatedMeasures",
+};
+
+/**
+ * Moves some attributes from `settings` to `permissions` and updates their name.
+ * Only moves an attribute if the corresponding permission doesn't exist yet.
+ * If there is no `permissions`, deletes the `settings` attribute that were supposed to be moved.
+ * Mutates `settings` and `permissions`.
+ */
+const moveSettingsToPermissions = (
+  settings: ContentRecord,
+  permissions: ContentRecord | undefined,
+): void => {
+  const settingsContent = JSON.parse(settings.entry.content);
+
+  if (!permissions) {
+    Object.keys(settingsToPermissionsMapping).forEach((settingKey) => {
+      if (settingKey in settingsContent) {
+        delete settingsContent[settingKey];
+      }
+    });
+
+    settings.entry.content = JSON.stringify(settingsContent);
+  } else {
+    const permissionsContent = JSON.parse(permissions.entry.content);
+
+    Object.entries(settingsToPermissionsMapping).forEach(
+      ([settingKey, permissionKey]) => {
+        if (settingKey in settingsContent) {
+          if (!(permissionKey in permissionsContent)) {
+            permissionsContent[permissionKey] = settingsContent[settingKey];
+          }
+          delete settingsContent[settingKey];
+        }
+      },
+    );
+
+    settings.entry.content = JSON.stringify(settingsContent);
+    permissions.entry.content = JSON.stringify(permissionsContent);
+  }
+};
+
+/**
+ * Transforms some settings attributes into permissions attributes.
+ * Applies the logic to the organization settings and permissions, as well as to each user's settings and permissions.
+ * Mutates `contentServer`.
+ */
+export const migrateSettings = (contentServer: ContentRecord): void => {
```

Comment:
Would `uiRootRecord` be a better name than `contentServer`? If yes, please also change it in `migrateSavedCalculatedMeasures` and all the other places.

Reply:
We call it `contentServer` because we actually pass the object describing the whole content server, and not only its `ui` part. Why were you thinking of `uiRootRecord` as a better name?
Note that we're using the term `contentServer` pretty much everywhere at the moment, so we're taking about the same thing, the change would be at a bigger scale than just the 5.0 --> 5.1 migration.

Reply:
Yeah my bad I read too fast. I would suggest the `rootRecord` name then instead of `contentServer` since the variable does not offer interaction with a content server: it's the record stored at its root.

Let's not do the change now since it's out of scope of this PR.